src/client/ua_client_subscriptions.c

```diff
@@ -111,61 +210,134 @@ UA_Client_Subscription_deleteInternal(UA_Client *client, UA_Client_Subscription
 
     /* temporary remove the subscriptions from the list */
-    for(size_t i = 0; i < request.subscriptionIdsSize; i++) {
-        subs[i] = findSubscription(client, request.subscriptionIds[i]);
-        if (subs[i])
-            LIST_REMOVE(subs[i], listEntry);
+    for (size_t i = 0; i < request->subscriptionIdsSize; i++) {
+        data->subs[i] = findSubscription(client, request->subscriptionIds[i]);
+        if (data->subs[i])
+            LIST_REMOVE(data->subs[i], listEntry);
     }
+    *retval = UA_STATUSCODE_GOOD;
+    return cc;
+cleanup:
+    *retval = UA_STATUSCODE_BADOUTOFMEMORY;
+    CustomCallback_remove(cc, false);
+    return NULL;
+}
 
-    /* Send the request */
-    UA_DeleteSubscriptionsResponse response;
-    __UA_Client_Service(client,
-                        &request, &UA_TYPES[UA_TYPES_DELETESUBSCRIPTIONSREQUEST],
-                        &response, &UA_TYPES[UA_TYPES_DELETESUBSCRIPTIONSRESPONSE]);
-    if(response.responseHeader.serviceResult != UA_STATUSCODE_GOOD)
+static void __Subscriptions_delete_handler(UA_Client *client, void *data, UA_UInt32 requestId, void *r) {
+    UA_DeleteSubscriptionsResponse *response = (UA_DeleteSubscriptionsResponse *)r;
+    CustomCallback *cc = (CustomCallback *)data;
+    Subscriptions_DeleteData *delData = (Subscriptions_DeleteData *)cc->clientData;
+    UA_DeleteSubscriptionsRequest *request = delData->request;
+    UA_Client_Subscription **subs = delData->subs;
+
+    if (response->responseHeader.serviceResult != UA_STATUSCODE_GOOD)
         goto cleanup;
 
-    if(request.subscriptionIdsSize != response.resultsSize) {
-        response.responseHeader.serviceResult = UA_STATUSCODE_BADINTERNALERROR;
+    if (request->subscriptionIdsSize != response->resultsSize) {
+        response->responseHeader.serviceResult = UA_STATUSCODE_BADINTERNALERROR;
         goto cleanup;
     }
 
     /* Loop over the removed subscriptions and remove internally */
-    for(size_t i = 0; i < request.subscriptionIdsSize; i++) {
-        if(response.results[i] != UA_STATUSCODE_GOOD &&
-           response.results[i] != UA_STATUSCODE_BADSUBSCRIPTIONIDINVALID) {
+    for (size_t i = 0; i < request->subscriptionIdsSize; i++) {
+        if (response->results[i] != UA_STATUSCODE_GOOD &&
+            response->results[i] != UA_STATUSCODE_BADSUBSCRIPTIONIDINVALID) {
             /* Something was wrong, reinsert the subscription in the list */
             if (subs[i])
                 LIST_INSERT_HEAD(&client->subscriptions, subs[i], listEntry);
+            subs[i] = NULL;
             continue;
         }
 
         if(!subs[i]) {
-            UA_LOG_INFO(&client->config.logger, UA_LOGCATEGORY_CLIENT,
-                        "No internal representation of subscription %u",
-                        request.subscriptionIds[i]);
+            UA_LOG_INFO(&client->config.logger, UA_LOGCATEGORY_CLIENT, "No internal representation of subscription %u",
```

Comment:
Keep lines short.
Avoid cosmetic changes in a feature-commit.

Reply:
That happened through old clang-format settings. I'm using the new clang-format config and now it doesnt make such changes. Will be fixed later